addons/mail/models/mail_thread.py

```diff
@@ -272,31 +272,35 @@ def create(self, vals_list):
 
         threads = super(MailThread, self).create(vals_list)
 
-        # automatic logging unless asked not to (mainly for various testing purpose)
-        if not self._context.get('mail_create_nolog'):
-            doc_name = self.env['ir.model']._get(self._name).name
-            for thread in threads:
-                thread._message_log(body=_('%s created') % doc_name)
-
         # auto_subscribe: take values and defaults into account
+        create_values_list = {}
         for thread, values in zip(threads, vals_list):
             create_values = dict(values)
             for key, val in self._context.items():
                 if key.startswith('default_') and key[8:] not in create_values:
                     create_values[key[8:]] = val
             thread._message_auto_subscribe(create_values)
+            create_values_list[thread.id] = create_values
 
-        # track values
+        # automatic logging unless asked not to (mainly for various testing purpose)
+        if not self._context.get('mail_create_nolog'):
+            doc_name = self.env['ir.model']._get(self._name).name
+            for thread in threads:
+                subtype = thread._creation_subtype()
+                body = _('%s created') % doc_name
+                if subtype:  # if we have a sybtype, post message to notify users from _message_auto_subscribe
+                    thread.sudo().message_post(body=body, subtype_id=subtype.id, author_id=self.env.user.partner_id.id, )
+                else:
+                    thread._message_log(body=body)  # todo optimise email_from
+
+        # post track template if a tracked field changed
         if not self._context.get('mail_notrack'):
-            if 'lang' not in self._context:
-                track_threads = threads.with_context(lang=self.env.user.lang)
-            else:
-                track_threads = threads
-            for thread, values in zip(track_threads, vals_list):
-                tracked_fields = thread._get_tracked_fields(list(values))
-                if tracked_fields:
-                    initial_values = {thread.id: dict.fromkeys(tracked_fields, False)}
-                    thread.message_track(tracked_fields, initial_values)
+            track_threads = threads.with_lang()
+            for thread in track_threads:
+                create_values = create_values_list[thread.id]
+                tracked_fields = self._get_tracked_fields(list(create_values))
+                changes = [field for field in tracked_fields if create_values.get(field)] # based on tracked field to stay consistent with write
```

Comment:
Shouldn't we keep void values and just do if field in create_values ? Indeed we may perform check on 0 / False values when deciding a subtype to trigger.

Reply:
Just maybe add a comment to say it is like old behavior and is actually wanted at create